gammapy/maps/axes.py

```diff
@@ -1885,6 +1945,32 @@ def __init__(self, edges_min, edges_max, reference_time, name="time", interp="li
         if np.any(delta < 0 * u.s):
             raise ValueError("Time intervals must not overlap.")
 
+    @property
+    def is_contiguous(self):
+        """Whether the axis is contiguous"""
+        return np.all(self.edges_min[1:] == self.edges_max[:-1])
+
+    @property
+    def contiguous(self):
+        """Make the time axis contiguous
+
+        Returns
+        -------
+        axis : `TimeMapAxis`
+            Contiguous time axis
+        """
+        values = np.unique(
+            [self.edges_min.to_value("s"), self.edges_max.to_value("s")]
```

Comment:
Does the conversion to second permit to not loose accuracy?
In the case of time_format='TimPlotDate' (see https://docs.astropy.org/en/stable/time/index.html#id3), the unit is days and the reference is  0001-01-01 00:00:00 UTC. So the year 2000 corresponds to 10^11 days, and if we add microseconds the data should contain a lot of digits. Python handles correctly these large accuracy?

Reply:
Good point, however `edges_min` and `edges_max` is always with respect to the reference time and by default all values are treated as `float64`, so it should be fine? See:
```python3
from astropy import units as u
import numpy as np

print(np.spacing((100 * u.year).to("microsecond")))
print(np.spacing((100 * u.year).to("second").astype(np.float32)))
```
Which prints:
```
0.5 us
256.0 s
```
So `float64` still reaches a precision of `0.5 us` in 100 years, while for `float32` it is only `256 s`.